### PR TITLE
Added link guidance content

### PR DESCRIPTION
**Pull request by bergerhoffer**
PR that rebased #28 and will continue addressing feedback from that PR in future updates here.

Preview: http://file.rdu.redhat.com/~ahoffer/2022/main-links.html#links

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@ndeevy Okay here is my attempt at wrapping up the link guidance from #28. Can you PTAL when you get a chance? 

I made the updates as separate commits so that you can see the progression of what changed compared to your initial PR. The second commit addresses some of the peer review feedback from the original PR. The third commit addresses the fact that we alphabetize the items in each category now. 

Here are a few of the significant changes I made:
* Updated the "internal link" section to be specifically about "cross-references". I was struggling with the idea of internal links meaning both cross-references and regular links to RH domain. So I thought it was better to get specific and keep this one just about xrefs. I got inspiration for this from Google's style guide: https://developers.google.com/style/cross-references
* Got rid of the general best practices section, because it was too hard to have a generic section, especially when I couldn't get it to come first alphabetically. I instead distributed the guidance to either the cross-references section or the external links section, as appropriate.
* Removed the references to not using JavaScript/iFrame/Flash links, because I don't even think this is possible in AsciiDoc

Let me know what you think and if there's anything you think I should revert or change. Thanks!

Preview: http://file.rdu.redhat.com/~ahoffer/2022/main-links.html#links

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (ndeevy) [inline supplementary_style_guide/style_guidelines/links.adoc:23]: @bergerhoffer you commented in the original PR about this and I agree with you, it's not very helpful or clear what it's trying to say. Maybe just ' Use hyperlinks' ?

**Review by ndeevy — DISMISSED** [review]

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: Thanks @ndeevy! I adjusted this to say to use hyperlinks unless the URL is an example URL or is otherwise inaccessible. 

I also removed the hyperlink guidance from the cross-references section (because those would always be hyperlinks), and instead replaced it with some guidance on including info right there instead of xreffing if it's critical.

PTAL at these updates and let me know what you think! Diff is here: https://github.com/redhat-documentation/supplementary-style-guide/pull/240/commits/b6ed97fb477f1e4f089e471eb928f238c89486e4

Comment (ndeevy) [reply supplementary_style_guide/style_guidelines/links.adoc]: Cleaner, clearer, awesome. LGTM

**Review by ndeevy — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@redhat-documentation/ccs-style-council This is a PR to wrap up the link guidance (originally in #28). PTAL when you can!

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bburt-rh) [inline supplementary_style_guide/style_guidelines/links.adoc:23]: Suggestion: Maybe consider adding `When linking to external sites, use URLs with top-level domain names, if possible.` E.g. `https://prometheus.io` vs `https://prometheus.io/docs/prometheus/latest/management_api/`. I'm not sure this is necessary, but it might help reduce broken external links.

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bburt-rh) [inline supplementary_style_guide/style_guidelines/links.adoc:24]: Consider cross-referencing the ISG guidance on sample URLs: the "Using addresses in examples" section here: https://www.ibm.com/docs/en/ibm-style?topic=elements-web-ip-email-addresses

**Comment by bburt-rh** [comment]
I added one suggestion to consider, but otherwise this lgtm. Thanks @bergerhoffer!

**Review by bburt-rh — APPROVED** [review]

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc:24]: @bburt-rh So we don't actually ever link to the ISG from the SSG (since the SSG is public, and the ISG is not available to everyone outside of RH). We also avoid naming the sections too, but if you think it's worthwhile I could add something like 

```
See the _IBM Style Guide_ for information about using addresses in examples.
```

wdyt?

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: Thanks, updated!

**Review by bburt-rh — DISMISSED** [review]

**Review by ndeevy — DISMISSED** [review]

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bredamc) [inline supplementary_style_guide/style_guidelines/links.adoc:41]: Perhaps we should define what is meant by "bare URL". I assumed that a bare URL is a link without link text, so I'm not sure if this point makes sense in this section, given that it's already included in the "External links" section? Perhaps it should be included in the Cross-references section instead?

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc:41]: @bredamc I actually thought about changing "bare URL" to "raw URL" because that's the term I'm more familiar with. wdyt? Or I could give an example if that would help.

I do think that the raw URL piece is more applicable in the external links section, since I don't even think it's possible to give a raw URL for cross-references.

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc:41]: @bergerhoffer Apologies for not being more clear -- we should absolutely retain that point in the "External links" section. I meant that we should include it in the "Cross-references" section instead of in the "Link text" section, but I agree that there's no need if it's not possible to provide a raw URL in a cross-reference anyway! 

Whether we call it a bare URL or a raw URL, I think we should define what it is. When we say "Do not use bare URLs", do we mean "Always include link text"? If so, should we phrase it in that way for clarity? If we retain this point in the "Link text" section, should that text be moved to be the first point?

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc]: @bergerhoffer Apologies for the delay.

I think we should rephrase the current "Bare URLs" text as follows:
"Always include link text. _Bare URLs_ (URLs without explanatory information) are unhelpful because they do not provide adequate context about the link target."

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc]: Examples are always good -- similar to the picture that paints a thousand words ;)

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: I think your suggested edit makes it crystal clear, thanks! Updated

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc]: One last tiny comment -- maybe reorder the list of points to list the "Do" statements first, and then the "Don't" statements?

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: Done!

**Review by bredamc — APPROVED** [review]
LGTM @bergerhoffer ! Many thanks for your patience :)

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (bburt-rh) [reply supplementary_style_guide/style_guidelines/links.adoc:25]: These are relatively common HTML hyperlink concepts that have been around for awhile and are easy to find more info about (as you discovered!). I guess it depends on the level of HTML knowledge we're assuming about the writers using the SSG. Is it typical that the average tech writer working in a docs-as-code, Asciidoc workflow would know these concepts? My view is "yes." Maybe other people have a different view...

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc:25]: I'm not familiar with those terms either, and would appreciate an explanation in the guide.

**Review by ndeevy — APPROVED** [review]

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (ndeevy) [reply supplementary_style_guide/style_guidelines/links.adoc:25]: This line is from the original guidance, which was for a much broader set of situations than docs. But I think you've got a point, Sharon and Breda, they're probably unknown to tech writers.
So maybe this line is not applicable to the documentation case really because we're being prescriptive about what to do in the rest of this guidance and writers are probably not in a position to use such types of links anyway? If that's the case, we could remove it.

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc:25]: So I just checked and apparently it is possible to define a link in AsciiDoc with a nofollow option. Would look something like this:

```
link:www.example.com[My example link, opts=nofollow]
```
So I think that this guidance is valuable to keep. I personally didn't know much about the difference of the two, other than knowing that you have to do something special to make a link "nofollow". 

However, I don't think we need to go too deep into explanations here. Maybe we could make it clearer that the default is followable/crawlable, and you'd have to do something special to enable "nofollow", and to not do that :)

What about something like this:

```suggestion
* By default, links are followable and crawlable. Do not use the "nofollow" link option unless absolutely necessary.
```

Comment (bredamc) [reply supplementary_style_guide/style_guidelines/links.adoc:25]: Your suggestion LGTM @bergerhoffer!

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: Thanks @bredamc, I pushed the change (so that if @bburt-rh and @sbmetz agree, this can be approved if otherwise ready).

Comment (ndeevy) [reply supplementary_style_guide/style_guidelines/links.adoc]: This makes sense!

**Review by bredamc — APPROVED** [review]
LGTM :)

File: supplementary_style_guide/style_guidelines/links.adoc
Comment (sbmetz) [reply supplementary_style_guide/style_guidelines/links.adoc]: That works for me! Thanks!

Comment (bergerhoffer) [reply supplementary_style_guide/style_guidelines/links.adoc]: Thanks all! @sbmetz would you mind approving the PR officially if it looks okay to you? (You go to files changed tab -> review changes -> hit the 'approve' radio button)

**Comment by bburt-rh** [comment]
Looks great! Thanks @bergerhoffer for all the iterations. LGTM.

**Review by bburt-rh — APPROVED** [review]

**Comment by bergerhoffer** [comment]
Has the necessary approvals, merging!

**Pull request merged**
